.github/workflows/test.yml

```diff
@@ -21,7 +21,8 @@ jobs:
       - name: install pymongo for 3.4 and 3.5
         run: |
           python -m pip install pymongo==3.11.0	
-        if: ${{ matrix.python-version == 3.4 || matrix.python-version == 3.5}} 
+        if: ${{ matrix.python-version == 3.4 || matrix.python-version == 3.5
+          || matrix.python-version == 2.7 }}
```

Comment:
Is this needed for Python 2.7? IIUC 2.7 will automatically install the latest <4.0 release (currently 3.12.3).

Reply:
I think you are right. I will remove it and if it breaks the tests I will add back in.

Reply:
It does break the tests for 2.7 if I remove this line.
```
Searching for pymongo>=3.4.0
Reading https://pypi.org/simple/pymongo/
Downloading https://files.pythonhosted.org/packages/3f/ff/7ec17064d403799d644f7db9c2e99202441849aebe771efe3199f3dd9076/pymongo-4.0.1.tar.gz#sha256=13d0624c13a91da71fa0d960205d93b3d98344481be865ee7cc238c972d41d73
Best match: pymongo 4.0.1
Processing pymongo-4.0.1.tar.gz
Writing /tmp/easy_install-I74Uv2/pymongo-4.0.1/setup.cfg
Running pymongo-4.0.1/setup.py -q bdist_egg --dist-dir /tmp/easy_install-I74Uv2/pymongo-4.0.1/egg-dist-tmp-p1yqWe
Traceback (most recent call last):
  File "setup.py", line 86, in <module>
    libraries=libraries
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/setuptools/__init__.py", line 145, in setup
    return distutils.core.setup(**attrs)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/distutils/core.py", line 151, in setup
    dist.run_commands()
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/setuptools/command/test.py", line 217, in run
    installed_dists = self.install_dists(self.distribution)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/setuptools/command/test.py", line 209, in install_dists
    tr_d = dist.fetch_build_eggs(dist.tests_require or [])
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/setuptools/dist.py", line 719, in fetch_build_eggs
    replace_conflicting=True,
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/pkg_resources/__init__.py", line 782, in resolve
    replace_conflicting=replace_conflicting
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1065, in best_match
    return self.obtain(req, installer)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1077, in obtain
    return installer(requirement)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/setuptools/dist.py", line 786, in fetch_build_egg
    return cmd.easy_install(req)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 679, in easy_install
    return self.install_item(spec, dist.location, tmpdir, deps)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 705, in install_item
    dists = self.install_eggs(spec, download, tmpdir)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 890, in install_eggs
    return self.build_and_install(setup_script, setup_base)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 1158, in build_and_install
    self.run_setup(setup_script, setup_base, args)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 1144, in run_setup
    run_setup(setup_script, args)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/setuptools/sandbox.py", line 253, in run_setup
    raise
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/contextlib.py", line 35, in __exit__
    self.gen.throw(type, value, traceback)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/setuptools/sandbox.py", line 195, in setup_context
    yield
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/contextlib.py", line 35, in __exit__
    self.gen.throw(type, value, traceback)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/setuptools/sandbox.py", line 166, in save_modules
    saved_exc.resume()
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/setuptools/sandbox.py", line 141, in resume
    six.reraise(type, exc, self._tb)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/setuptools/sandbox.py", line 154, in save_modules
    yield saved
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/setuptools/sandbox.py", line 195, in setup_context
    yield
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/setuptools/sandbox.py", line 250, in run_setup
    _execfile(setup_script, ns)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/setuptools/sandbox.py", line 45, in _execfile
    exec(code, globals, locals)
  File "/tmp/easy_install-I74Uv2/pymongo-4.0.1/setup.py", line 9, in <module>
    # Unless required by applicable law or agreed to in writing, software
RuntimeError: Python version >= 3.6 required.
```

Reply:
Ah the problem in that `tests_require` using easy_install to fetch and install pymongo. Instead we should just always install pymongo via pip for all python versions:
```
      - name: install pymongo
        run: |
          python -m pip install pymongo  
```

This _should_ work for all python versions as long as pip is up to date. Can you test this one out?

Reply:
Yes. Just tried that out and it works.